lib/internal/bootstrap/pre_execution.js

```diff
@@ -477,6 +465,9 @@ function shouldUseESMLoader(mainPath) {
 }
 
 function runMainESM(mainPath) {
+  process.emitWarning(
+    'The ESM module loader is experimental.',
+    'ExperimentalWarning', undefined);
```

Comment:
This warning is not printed if the module graph is entered with a dynamic import from CommonJS

Reply:
Thanks, I've fixed this up to apply lazily for dynamic import.